File: desktop/src/shared/types/quickNote.ts
import type { ApiListData } from './api';

export type QuickNoteStatus = 'active' | 'deleted';

// TODO: Align source device enum with backend once multi-device sync contracts are defined.
export type QuickNoteSourceDevice = 'desktop';

export type QuickNoteListStatus = 'ok' | 'empty' | 'readonly' | 'error';

export interface QuickNote {
  id: string;
  date: string;
  content: string;
  sourceDevice: QuickNoteSourceDevice;
  status: QuickNoteStatus;
  createdAt: string;
  updatedAt: string;
}

export interface QuickNotePermissions {
  canView: boolean;
  canCreate: boolean;
  canUpdate: boolean;
  canDelete: boolean;
  reason?: string;
}

export interface QuickNoteListQuery {
  date: string;
  pageNo?: number;
  pageSize?: number;
  status?: QuickNoteStatus;
}

export interface CreateQuickNotePayload {
  content: string;
  date?: string;
}

export interface UpdateQuickNotePayload {
  id: string;
  content: string;
}

export type QuickNoteListData = ApiListData<QuickNote, QuickNotePermissions> & {
  status: QuickNoteListStatus;
  permissions: QuickNotePermissions;
};

File: mobile/lib/features/daily_markdown/domain/quick_note.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'quick_note.freezed.dart';

/// 快速记录。字段对齐桌面端 `shared/types/quickNote.ts`。
/// `sourceDevice` 固定 `'desktop'`（与桌面端一致，多端枚举待后续统一）。
@freezed
class QuickNote with _$QuickNote {
  const factory QuickNote({
    required String id,
    required String date,
    required String content,
    @Default('desktop') String sourceDevice,
    @Default('active') String status,
    required String createdAt,
    required String updatedAt,
  }) = _QuickNote;
}
